src/Raven.Studio/typescript/components/pages/database/documents/identities/DocumentIdentities.tsx

```diff
@@ -0,0 +1,122 @@
+
+    const identitiesTable = useReactTable({
+        columns: identitiesColumns,
+        data: identities,
+        columnResizeMode: "onChange",
+        getCoreRowModel: getCoreRowModel(),
+        getSortedRowModel: getSortedRowModel(),
+        getFilteredRowModel: getFilteredRowModel(),
+        onGlobalFilterChange: setGlobalFilter,
+        globalFilterFn: "includesString",
+        filterFns: {},
+        state: {
+            globalFilter,
+        },
+    });
+
+    const debouncedSearch = debounce((e: ChangeEvent<HTMLInputElement>) => setGlobalFilter(e.target.value), 300);
+
+    return (
+        <>
+            <div className="content-margin">
+                <Col xs={12} md={12} lg={7}>
+                    <AboutViewHeading title="Identities" icon="identities" />
+                    <Row className="justify-content-between rows-cols-lg-2 row-cols-1 mb-4">
+                        <Col md={12} lg={8} xl={9}>
+                            <Button
+                                color="primary"
+                                onClick={() => setIsOpen(true)}
+                                disabled={!databaseAccessWrite}
+                                className="add-new-identity-btn py-2"
+                            >
+                                <Icon icon="plus" className="mr-2" />
+                                Add new identity
+                            </Button>
+                        </Col>
+                        <Col md={12} lg={4} xl={3}>
+                            <Input onChange={debouncedSearch} placeholder="Filter prefix" />
+                        </Col>
+                    </Row>
+
+                    <Card>
+                        <CardBody>
+                            <VirtualTable
+                                table={identitiesTable}
+                                className="mt-3"
+                                isLoading={loading}
+                                heightInPx={800}
+                            />
+                        </CardBody>
+                    </Card>
+                </Col>
+                <Col xs={12} md={12} lg={5}>
+                    <DocumentIdentitiesAboutView />
+                </Col>
+            </div>
+            <DocumentIdentitiesAddModal isOpen={isOpen} toggleModal={toggleModal} />
+        </>
+    );
+}
+
+function useGetIdentities(database: string) {
+    const [identities, setIdentities] = useState<Identity[]>([]);
+    const { databasesService } = useServices();
+    const asyncGetIdentities = useAsync(() => databasesService.getIdentities(database), [], {
+        onSuccess: (result) =>
+            setIdentities(
+                Object.keys(result)
+                    .map((identity) => ({
+                        prefix: identity,
+                        value: result[identity],
+                    }))
+                    .filter(Boolean)
+            ),
+    });
```

Comment:
`useAsync` already stores result so you don't need to do it. It is in `asyncGetIdentities.result`.
```suggestion
    const asyncGetIdentities = useAsync(async () => {
        const result = await databasesService.getIdentities(database);
        
        return Object.keys(result)
                    .map((identity) => ({
                        prefix: identity,
                        value: result[identity],
                    }));
    }, []);
```